src/backend/data/feed.js

```diff
@@ -13,7 +13,7 @@ const hash = require('./hash');
 const urlToId = url => hash(normalizeUrl(url));
 
 class Feed {
-  constructor(author, url, etag, lastModified) {
+  constructor(author, url, user, etag, lastModified) {
```

Comment:
I read this PR last night on my phone, and noticed that you had some ordering bugs between callers and this constructor.  It looks like you've maybe fixed them all.

One tip for doing a change like this.  Start with a simple test that calls the constructor and makes sure that all the arguments passed in get set on the correct properties of the resulting object.  That will catch arguments being in the wrong order.

Next, use git to help you find all the callers:

```
$ git grep "new Feed"
src/backend/data/feed.js:   * Creates a new Feed object by extracting data from the given feed-like object.
src/backend/data/feed.js:    const feed = new Feed(feedData.author, feedData.url, feedData.etag, feedData.lastModified);
src/backend/data/feed.js:    return new Feed(data.author, data.url, data.etag, data.lastModified);
src/backend/web/routes/feed.js:  const feed = new FeedMeta({
test/feed.test.js:  const createFeed = () => new Feed(data.author, data.url);
test/feed.test.js:    expect(() => new Feed('author', undefined)).toThrow();
test/feed.test.js:    expect(() => new Feed(undefined, undefined)).toThrow();
test/feed.test.js:      const feed = new Feed('author', 'http://url.com/with/etag');
test/feed.test.js:      const feed = new Feed('author', 'http://url.com/with/lastModified');
test/feeds-wiki.test.js:    await new Feed(testFeedAuthor, testFeedURL).save();
test/feeds.test.js:    return new Feed('foo', `http://telescope${item}.cdot.systems`);
test/feeds.test.js:  const addedFeed = new Feed('foo', existingUrl);
test/posts.test.js:  const feed1 = new Feed('Feed Author', 'http://feed-url.com/', null, null);
test/storage.test.js:  const feed1 = new Feed('James Smith', 'http://seneca.co/jsmith');
test/storage.test.js:  const feed2 = new Feed('James Smith 2', 'http://seneca.co/jsmith/2');
test/storage.test.js:  const feed3 = new Feed('James Smith 2', 'http://seneca.co/jsmith/3', 'etag');
test/storage.test.js:  const feed4 = new Feed('James Smith 2', 'http://seneca.co/jsmith/4', 'etag', 'last-modified');
```

Then you can work through the list and correct them all.

Finally, once you have a long list of things to pass to a constructor like this, it's often wise to switch to an `options` Object pattern: pass 1 argument to the constructor with all the desired data as properties.  This has the advantage of not having to deal with ordering.

Reply:
Ah - should I implement this?

Reply:
I would leave it as is, but fix your call ordering.  If you refactor it to use an Object, it will touch a lot more code than what you already are doing.

Reply:
Okay - I think I'm good to resolve this one, for now, if we continue to make the user object larger, we should switch to passing an object through to the constructor opposed too all the different parameters.